src/plugins/section_title.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * © 2024 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { schema } from '@manuscripts/transform'
+import { Node as ProseMirrorNode } from 'prosemirror-model'
+import { EditorState, Plugin, PluginKey, Transaction } from 'prosemirror-state'
+import { findChildrenByType } from 'prosemirror-utils'
+
+type NumberingArray = number[]
+export type PluginState = Map<string, string>
+
+export const sectionTitleKey = new PluginKey<PluginState>('sectionNumbering')
+
+const calculateSectionLevels = (
+  node: ProseMirrorNode,
+  startPos: number,
+  numbering: NumberingArray = [0]
+) => {
+  const sectionNumberMap = new Map<string, string>()
+  node.forEach((childNode, offset) => {
+    if (childNode.type === schema.nodes.section) {
```

Comment:
Do we need to exclude "deleted" sections from the order?

Reply:
yes, I have included a check for the deleted and rejected sections.
as well as, I have introduce new function to handle the section_title attribute, because when one of the sections got reverted or deleted the updateContent method will be triggered only for that node. so I have added the function to loop over the dom and update the attributes accordingly.